src/wasm-binary.h

```diff
@@ -116,6 +116,7 @@ struct LEB {
   }
 };
 
+#define SIZE32LEB(v) U32LEB((uint32_t)(v))
```

Comment:
What about just changing the constructor of LEB to take a `size_t` argument instead of a `T`? That would force all arguments to be implicitly promoted at the construction site when necessary, and then the implementation would cast down to `T` and could also assert that it doesn't overflow.